waku/handshake.go

```diff
@@ -87,80 +70,9 @@ func (o statusOptions) PoWRequirementF() *float64 {
 	return &result
 }
 
-func (o *statusOptions) SetPoWRequirementFromF(val float64) {
-	requirement := math.Float64bits(val)
-	o.PoWRequirement = &requirement
-}
-
-func (o statusOptions) EncodeRLP(w io.Writer) error {
-	v := reflect.ValueOf(o)
-	var optionsList []interface{}
-	for i := 0; i < v.NumField(); i++ {
-		field := v.Field(i)
-		if !field.IsNil() {
-			value := field.Interface()
-			key, ok := idxFieldKey[i]
-			if !ok {
-				continue
-			}
-			if value != nil {
-				optionsList = append(optionsList, []interface{}{key, value})
-			}
-		}
-	}
-	return rlp.Encode(w, optionsList)
-}
-
-func (o *statusOptions) DecodeRLP(s *rlp.Stream) error {
-	_, err := s.List()
-	if err != nil {
-		return fmt.Errorf("expected an outer list: %v", err)
-	}
-
-	v := reflect.ValueOf(o)
-
-loop:
-	for {
-		_, err := s.List()
-		switch err {
-		case nil:
-			// continue to decode a key
-		case rlp.EOL:
-			break loop
-		default:
-			return fmt.Errorf("expected an inner list: %v", err)
-		}
-		var key string
-		if err := s.Decode(&key); err != nil {
-			return fmt.Errorf("invalid key: %v", err)
-		}
-		// Skip processing if a key does not exist.
-		// It might happen when there is a new peer
-		// which supports a new option with
-		// a higher index.
-		idx, ok := keyFieldIdx[key]
-		if !ok {
-			// Read the rest of the list items and dump them.
-			_, err := s.Raw()
-			if err != nil {
-				return fmt.Errorf("failed to read the value of key %s: %v", key, err)
-			}
-			continue
-		}
-		if err := s.Decode(v.Elem().Field(idx).Addr().Interface()); err != nil {
-			return fmt.Errorf("failed to decode an option %s: %v", key, err)
-		}
-		if err := s.ListEnd(); err != nil {
-			return err
-		}
-	}
-
-	return s.ListEnd()
-}
-
 func (o statusOptions) Validate() error {
-	if len(o.TopicInterest) > 1000 {
-		return errors.New("topic interest is limited by 1000 items")
+	if len(o.TopicInterest) > 10000 {
+		return errors.New("topic interest is limited by 10000 items")
 	}
 	return nil
 }
```

Comment:
Does `waku/1` introduce a higher topic limit?

Reply:
Already `waku/0` should have had the higher topic limit https://github.com/vacp2p/specs/blob/master/specs/waku/waku.md , I think it was left off from the implementation.